src/main/java/org/jabref/logic/importer/fetcher/AdvancedSearchConfig.java

```diff
@@ -0,0 +1,133 @@
+package org.jabref.logic.importer.fetcher;
+
+import org.jabref.model.strings.StringUtil;
+
+public class AdvancedSearchConfig {
+    private final String defaultField;
+    private final String author;
+    private final String title;
+    private final int fromYear;
+    private final int toYear;
+    private final String journal;
+
+    AdvancedSearchConfig(String defaultField, String author, String title, int fromYear, int toYear, String journal) {
+        this.defaultField = defaultField;
+        this.author = author;
+        this.title = title;
+        this.fromYear = fromYear;
+        this.toYear = toYear;
+        this.journal = journal;
+    }
+
+    public String getDefaultField() {
+        return defaultField;
+    }
+
+    public String getAuthor() {
+        return author;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public int getFromYear() {
+        return fromYear;
+    }
+
+    public int getToYear() {
+        return toYear;
+    }
+
+    public String getJournal() {
+        return journal;
+    }
+
+    public static AdvancedSearchConfigBuilder builder() {
+        return new AdvancedSearchConfigBuilder();
+    }
+
+    public static class AdvancedSearchConfigBuilder {
+        private String defaultField = "";
+        private String author = "";
+        private String title = "";
+        private int fromYear;
+        private int toYear;
+        private String journal = "";
+
+        public AdvancedSearchConfigBuilder() {
+        }
+
+        /**
+         * @param anyField Search string for any field
+         * @return Builder instance
+         */
+        public AdvancedSearchConfigBuilder anyField(String anyField) {
+            this.defaultField = anyField;
```

Comment:
Please gard all sets in the builder with [Objects.requireNonNull](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-). This way, trivial, but nasty, bugs can be found.

```suggestion
            this.defaultField = Objects.requireNonNull(anyField);
```